backend/src/models/Table.js

```diff
@@ -0,0 +1,79 @@
+const db = require('./db');
+
+function validate(table) {
+  if (table.tableName === undefined) {
+    throw Error(message='DB table name is not set');
+  };
+  if (table.pk === undefined) {
+    throw Error(message='PK name is not set');
+  };
+  if (table[table.pk] === undefined) {
+    throw Error(message=`'${table.pk}' is not set`);
+  }
+}
+
+class Table {
+  constructor(tableName, pk, data) {
+    this.tableName = tableName;
+    this.pk = pk;
+    if (pk in data) {
+      this[pk] = data[pk];
+      delete data[pk];
+    };
+    this.data = data;
+  }
+
+  create() {
+    validate(this);
+    let index = 1;
+    const prepared = Object.keys(this.data).reduce((accumulator, key) => {
+            accumulator.keys.push(key);
+            accumulator.indexes.push(`$${index++}`);
+            accumulator.values.push(this.data[key]);
+            return accumulator;
+          }, {keys: [], indexes: [], values: []});
+    prepared.values.push(this[this.pk]);
+    const text = `INSERT INTO ${this.tableName} (${prepared.keys.join(', ')}, ${this.pk}) VALUES (${prepared.indexes.join(', ')}, $${index});`;
+    // eventually it will result in:
+    // text: INSERT INTO users (first_name, last_name, password, email) VALUES ($1, $2, $3, $4);
+    // prepared.values: ['Kenny', 'McCormick', '$2b$10$XBUfi5ztRo0EKQ5XaOVEf.KXdU8km1.SFm9fybKE3bjk8L5qTVMNe', 'kenny@gmail.com']
+    return db.query(text, prepared.values);
```

Comment:
I think we need a `catch` block here to propagate the error to the caller. Same goes for the other methods that call `db.query`

Reply:
That's a promise, It will pop-up later in catch block in router, and thus give us the opportunity to return error message with `res.json(...)`